cyberdrop_dl/scraper/crawlers/e621_crawler.py

```diff
@@ -0,0 +1,140 @@
+
+    from cyberdrop_dl.managers.manager import Manager
+
+
+class E621Crawler(Crawler):
+    primary_base_domain = URL("https://e621.net")
+
+    def __init__(self, manager: Manager) -> None:
+        super().__init__(manager, "e621.net", "E621")
+        self.custom_headers = {"User-Agent": f"CyberDrop-DL/{__version__} (by B05FDD249DF29ED3)"}
+        self.request_limiter = AsyncLimiter(2, 1)
+
+    """~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"""
+
+    @create_task_id
+    async def fetch(self, scrape_item: ScrapeItem) -> None:
+        """Determines where to send the scrape item based on the URL."""
+
+        if "tags" in scrape_item.url.query_string:
+            await self.tag(scrape_item)
+        elif scrape_item.url.path.startswith("/posts/"):
+            await self.file(scrape_item)
+        elif scrape_item.url.path.startswith("/pools/"):
+            await self.pool(scrape_item)
+        else:
+            raise ValueError("Invalid e621 URL format")
+
+    async def paginator(self, scrape_item: ScrapeItem) -> AsyncGenerator[dict]:
+        """Generator for album pages."""
+        page = int(scrape_item.url.query.get("page", 1))
+        while True:
+            async with self.request_limiter:
+                params = {"tags": scrape_item.url.query["tags"], "page": page}
+                response = await self.client.get_json(
+                    self.domain,
+                    self.primary_base_domain / "posts.json",
+                    params=params,
+                    origin=scrape_item,
+                    headers_inc=self.custom_headers,
+                )
+            posts = response.get("posts", [])
+            yield posts
+            if posts:
+                page += 1
+                continue
+            break
+
+    @error_handling_wrapper
+    async def tag(self, scrape_item: ScrapeItem) -> None:
+        """Fetches posts from e621 based on tags."""
+        tags = scrape_item.url.query["tags"]
+        scrape_item.set_type(FILE_HOST_ALBUM, self.manager)
+        title = self.create_title(tags.replace("+", " "))
+        scrape_item.part_of_album = True
+
+        async for response in self.paginator(scrape_item):
+            for post in response:
+                file_url = post.get("file", {}).get("url")
+                if not file_url:
+                    continue
+
+                creation_date = post.get("created_at")
+                timestamp = self.parse_datetime(creation_date)
+
+                link = self.parse_url(file_url)
+                new_scrape_item = self.create_scrape_item(
+                    scrape_item, link, title, True, add_parent=scrape_item.url, possible_datetime=timestamp
+                )
+                scrape_item.add_children()
+                filename, ext = get_filename_and_ext(link.name)
+                await self.handle_file(link, new_scrape_item, filename, ext)
```

Comment:
Increase children count after the the file is processed
```suggestion
                filename, ext = get_filename_and_ext(link.name)
                await self.handle_file(link, new_scrape_item, filename, ext)
                scrape_item.add_children()
```